code/tracing/MainFrameTimer.cpp

```diff
@@ -0,0 +1,34 @@
+//
+//
+
+#include "MainFrameTimer.h"
+
+#include "tracing/scopes.h"
+#include "tracing/categories.h"
+
+namespace tracing {
+
+MainFrameTimer::MainFrameTimer() : _out("profiling.csv") {
+}
+MainFrameTimer::~MainFrameTimer() {
+	_out.close();
+}
+void MainFrameTimer::processEvent(const trace_event* event) {
+	if (event->scope != &MainFrameScope || event->category != &MainFrame) {
+		return;
+	}
+
+	if (event->type == EventType::AsyncBegin) {
```

Comment:
Can use a switch block here instead of if-else blocks if `event->type` is the only thing used